lib/vrl/compiler/src/compiler.rs

```diff
@@ -607,6 +609,81 @@ impl<'a> Compiler<'a> {
         Some(target)
     }
 
+    #[cfg(feature = "expr-function_call")]
+    fn check_metadata_function_deprecations(&mut self, func: &FunctionCall, args: &ArgumentList) {
+        if func.ident == "get_metadata_field" {
+            if let Ok(key) = get_metadata_key(args) {
+                match key {
+                    MetadataKey::Query(target_path) => self.diagnostics.push(Box::new(
+                        DeprecationWarning::new("the \"get_metadata_field\" function")
+                            .with_span(func.span)
+                            .with_notes(Note::solution(
+                                "using the metadata path syntax instead",
+                                vec![format!("{}", target_path)],
+                            )),
+                    )),
+                    MetadataKey::Legacy(secret_key) => self.diagnostics.push(Box::new(
+                        DeprecationWarning::new("the \"get_metadata_field\" function")
+                            .with_span(func.span)
+                            .with_notes(Note::solution(
+                                "using the \"get_secret\" function instead",
+                                vec![format!("get_secret(\"{}\")", secret_key)],
+                            )),
+                    )),
+                }
+            }
+        }
+        if func.ident == "set_metadata_field" {
```

Comment:
Pedantic suggestion, only fix in case there's other stuff to amend:

```suggestion
        }

        if func.ident == "set_metadata_field" {
```